src/main/java/edu/cornell/kfs/coa/businessobject/CuAccountGlobal.java

```diff
@@ -2,164 +2,824 @@
-                }
-    
-                // FINANCIAL HIGHER ED FUNCTION CODE
-                if (StringUtils.isNotBlank(financialHigherEdFunctionCd)) {
-                    account.setFinancialHigherEdFunctionCd(financialHigherEdFunctionCd);
-                }
-    
-                // SUFFICIENT FUNDS CODE
-                if (StringUtils.isNotBlank(accountSufficientFundsCode)) {
-                    account.setAccountSufficientFundsCode(accountSufficientFundsCode);
-                }
-    
-                // LABOR BENEFIT RATE CATEGORY CODE
-                if (StringUtils.isNotBlank(getLaborBenefitRateCategoryCode())) {
-                    account.setLaborBenefitRateCategoryCode(getLaborBenefitRateCategoryCode());
-                }
-    
-                // PENDING ACCOUNT SUFFICIENT FUNDS CODE INDICATOR
-                if (pendingAcctSufficientFundsIndicator != null) {
-                    account.setPendingAcctSufficientFundsIndicator(pendingAcctSufficientFundsIndicator);
-                }
-    
-                // MAJOR REPORTING CATEGORY CODE 
-                if (StringUtils.isNotBlank(majorReportingCategoryCode)) {
-                    ((AccountExtendedAttribute) account.getExtension()).setMajorReportingCategoryCode(majorReportingCategoryCode);
-                }
-                
-                
-                persistables.add(account);
+    public CuAccountGlobal() {
+    	super();
+    	indirectCostRecoveryAccounts = new ArrayList<IndirectCostRecoveryAccountChange>();
+	}
+
     
-            }
-        }
+	@Override
+    public List<PersistableBusinessObject> generateGlobalChangesToPersist() {
+        List<PersistableBusinessObject> persistables = new ArrayList<PersistableBusinessObject>();
 
+        for (AccountGlobalDetail accountGlobalDetail : accountGlobalDetails) {
+        	updateAccountValuesAndAddToPersistablesList(persistables, accountGlobalDetail);
+        }
+        
         return persistables;
     }
+
+	private void updateAccountValuesAndAddToPersistablesList(
+			List<PersistableBusinessObject> persistables,
+			AccountGlobalDetail accountGlobalDetail) {
+		Account account = updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccountGlobalDoc(accountGlobalDetail);
+
+		if (ObjectUtils.isNotNull(account)) {
+			persistables.add(account);
+		}
+	}
+    
+	private Account updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccountGlobalDoc(
+			GlobalBusinessObjectDetailBase globalDetail) {
+		AccountGlobalDetail accountGlobalDetail = (AccountGlobalDetail) globalDetail;
+		Account account = SpringContext.getBean(BusinessObjectService.class).findByPrimaryKey(Account.class,accountGlobalDetail.getPrimaryKeys());
+
+		if (ObjectUtils.isNotNull(account)) {
+			updateAccountBasicFields(account);
+			updateAccountExtendedAttribute(account);
+			
+			if(ObjectUtils.isNotNull(this.getIndirectCostRecoveryAccounts()) && this.getIndirectCostRecoveryAccounts().size() > 0){
```

Comment:
The updateIcrAccounts method will update the ICR accounts collection on the account being edited by the global document with values entered on the account global doc. The ICR collection on the account global doc is this.getIndirectCostRecoveryAccounts(). If this is null or empty it means no ICR accounts have been entered on the Account Global document so there is no need to attempt to update ICR accounts on the account being edited.